src/Knp/DictionaryBundle/Command/DictionaryDumpCommand.php

```diff
@@ -83,12 +85,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
     private function getDictionariesDetail($dictionaryName = null)
     {
         $tableRows = [];
-        /** @var Dictionary $dico */
-        foreach ($this->registry->all() as $dico) {
-            if (!\is_null($dictionaryName) && $dictionaryName === $dico->getName() || \is_null($dictionaryName)) {
-                $tableRows[] = ["<fg=cyan>{$dico->getName()}</fg=cyan>"];
+
+        if (!\is_null($dictionaryName)) {
+            $dico = $this->registry->get($dictionaryName);
+            foreach ($dico as $key => $value) {
```

Comment:
What do you think about this duplication of code ? Maybe should we instanciate `$dico` outside of the `getDictionariesDetails` and pass it as parameters like below ?
```php
protected function execute(InputInterface $input, OutputInterface $output)
{
    // ...

    // Read dictionaries information
    if (\is_null($dictionaryName)) {
        $dictionaries = $this->registry->all();
    } else {
        $dictionaries = [$this->registry->get($dictionaryName)];
    }
    $tableRows = $this->getDictionariesDetail($dictionaries);

    // ...
}

private function getDictionariesDetail($dictionaries)
{
    $tableRows = [];
    foreach ($dictionaries as $dico) {
        foreach ($dico as $key => $value) {
            $tableRows[$dico->getName()][] = [$key, $value];
       }
    }

    return $tableRows;
}
```

Reply:
Yep, good idea !

Reply:
Or instanciate it before the foreach, as you want ;)